### PR TITLE
Add support for shm-size.

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@umohnani8 @baude @mheon @TomSweeneyRedHat PTAL

File: libpod/sql_state_internal.go [hunk 1]
```diff
@@ -170,6 +170,7 @@ func prepareDB(db *sql.DB) (err error) {
             MountLabel TEXT NOT NULL,
             Mounts TEXT NOT NULL,
             ShmDir TEXT NOT NULL,
+            ShmSize TEXT NOT NULL,
```
Comment (TomSweeneyRedHat) [inline libpod/sql_state_internal.go]: Is TEXT right here and string right below?  Or should they be INTEGER and int?

Comment (rhatdan) [reply libpod/sql_state_internal.go]: THey are passed around and used as a string.  You can use indicators in the string like
52M or 65k

**Comment by mheon** [comment]
NACK for now, I am working on DB changes to include SHM size and all the other things that need DB modifications now, please wait for that and rebase on top of it.

**Comment by rhatdan** [comment]
@mheon Updated to use your new database. PTAL.

File: libpod/container.go [hunk 2]
```diff
@@ -686,12 +685,32 @@ func (c *Container) Init() (err error) {
...
+	mnt := spec.Mount{
+		Type:        "bind",
+		Source:      c.config.ShmDir,
+		Destination: "/dev/shm",
+		Options:     []string{"rw", "bind"},
+	}
+	g.AddMount(mnt)
 	// Bind mount resolv.conf
-	g.AddBindMount(runDirResolv, "/etc/resolv.conf", []string{"rw"})
+	mnt = spec.Mount{
```
Comment (mheon) [inline libpod/container.go]: Can you name this something different? It's a little bit confusing using mnt for 3 different mounts in the same function

File: libpod/container.go [hunk 3]
```diff
@@ -686,12 +685,32 @@ func (c *Container) Init() (err error) {
...
+		Type:        "bind",
+		Source:      runDirResolv,
+		Destination: "/etc/resolv.conf",
+		Options:     []string{"rw", "bind"},
+	}
+	g.AddMount(mnt)
 	// Bind mount hosts
-	g.AddBindMount(runDirHosts, "/etc/hosts", []string{"rw"})
+	mnt = spec.Mount{
```
Comment (mheon) [inline libpod/container.go]: Same as above - can you rename this variable?

File: libpod/options.go [hunk 4]
```diff
@@ -283,14 +283,15 @@ func WithCNIPluginDir(dir string) RuntimeOption {
 
 // Container Creation Options
 
-// WithShmDir sets the directory that should be mounted on /dev/shm
-func WithShmDir(dir string) CtrCreateOption {
+// WithShmDir sets the directory that should be mounted on /dev/shm and the size of it
+func WithShmDir(dir string, size int64) CtrCreateOption {
```
Comment (mheon) [inline libpod/options.go]: Can you separate size into a separate option? I can see people wanting the default location we use but a custom size or vice versa

File: libpod/runtime_ctr.go [hunk 5]
```diff
@@ -38,6 +38,7 @@ func (r *Runtime) NewContainer(rSpec *spec.Spec, options ...CtrCreateOption) (c
 		return nil, err
 	}
 	ctr.config.StopTimeout = CtrRemoveTimeout
+	ctr.config.ShmSize = DefaultShmSize
```
Comment (mheon) [inline libpod/runtime_ctr.go]: This is already set in newContainer in container.go

**Comment by mheon** [comment]
Small question: why are we customizing a lot of mounts now in the spec - did something change?

**Comment by rhatdan** [comment]
@mheon AddBindMount,AddCgroupMount,AddTmpfsMount were dropped, and you are now supposed to setup your own mount spec and pass it into AddMount

File: libpod/runtime_ctr.go (hunk 5 above)
Comment (rhatdan) [reply libpod/runtime_ctr.go]: Removed

File: libpod/options.go (hunk 4 above)
Comment (rhatdan) [reply libpod/options.go]: Done

File: libpod/container.go (hunk 3 above)
Comment (rhatdan) [reply libpod/container.go]: I did, although not sure it adds much value.

File: libpod/container.go (hunk 2 above)
Comment (rhatdan) [reply libpod/container.go]: Done

**Comment by mheon** [comment]
Ah, that explains it. Thanks!

**Comment by mheon** [comment]
LGTM

**Comment by rhatdan** [comment]
@TomSweeneyRedHat @umohnani8 @baude PTAL

**Comment by rhatdan** [comment]
@rh-atomic-bot r=TomSweeneyRedHat

**Pull request closed without merging**